src/models/searchservice/response/DirectAnswer.ts

```diff
@@ -6,7 +6,7 @@ import { DirectAnswerType } from './DirectAnswerType';
  *
  * @public
  */
-export interface DirectAnswer {
+export interface DirectAnswer<T = unknown> {
```

Comment:
I think Yen wanted to leave the type as string for FieldType here as it is possible that we get other types, and then as we add specific Direct Answers (AddressDirectAnswers, PhoneDirectAnswers...) we would use the enum fields.

Reply:
I think we do want to have `fieldType` use the new `FieldType` enum here. Davish let me know there are different types of featured snippet as well. So, I think we'd use `fieldType` there too to discriminate between the different types of snippet DA.

Reply:
I was wondering if we needed to worry about `FieldType` being a union type string vs an enum. long story short enum should be good! My primary concern was whether user's could switch on a type that is either enum or string so I wrote a short test
```ts
enum OnionTypes {
  Sweet = 'sweet',
  Red = 'Red',
  Negi = 'negi!!!!'
}

type Onion = OnionTypes | string;

function handle(o: Onion) {
  switch (o) {
    case 'sweet':
      console.log('sweet', o);
      break;
    case OnionTypes.Negi:
      console.log('negi', o);
      break;
    default:
      console.log('default', o);
  }
}

it('ogres are like onions', async () => {
  handle(OnionTypes.Sweet);
  handle(OnionTypes.Negi);
  handle('unknown onion');
});
```

Reply:
but yeah @juliannzhou we'll probably want fieldType to be `fieldType: FieldType | string`

Reply:
I was thinking we could use the same `EnumOrLiteral` concept we had in headless so it would accept FieldType in enum or string form, instead of doing `| string`?

Reply:
maybe it's a recent typescript change but it's possible we don't need EnumOrLiteral anymore? It looks like typescript automatically does something like EnumOrLiteral now? 
![image](https://user-images.githubusercontent.com/23005393/184181263-ecf84f40-68ab-4f9e-bf04-01591c671ce2.png)

Reply:
I'm not sure if I understand but it looks like if you type a string form of an enum value, it wouldn't work?
<img width="638" alt="Screen Shot 2022-08-11 at 12 24 26 PM" src="https://user-images.githubusercontent.com/36055303/184182166-c1df178a-3c72-4995-9cd7-fc1138b814a9.png">

Reply:
I looked through a couple releases and didn't see anything, do you think you could dig a little bit @yen-tt to see if we need EnumOrLiteral still?

Reply:
I looked through a couple releases and didn't see anything, but I can dig a little deeper to check if we still need EnumOrLiteral

Reply:
that makes sense, I'm down to use `EnumOrLiteral` here!